src/zenml/integrations/github/code_repositories/github_code_repository.py

```diff
@@ -69,10 +70,43 @@ def github_repo(self) -> Repository:
 
         Returns:
             The GitHub repository.
+
+        Raises:
+            RuntimeError: If the repository cannot be found.
         """
-        return self._github_session.get_repo(
-            f"{self.config.owner}/{self.config.repository}"
-        )
+        try:
+            github_repository = self._github_session.get_repo(
+                f"{self.config.owner}/{self.config.repository}"
+            )
+        except GithubException as e:
+            raise RuntimeError(
+                f"An error occurred while getting the repository: {str(e)}"
+            )
+        return github_repository
+
+    def check_github_repo_public(self, owner, repo) -> None:
+        """Checks if a GitHub repository is public.
+
+        Args:
+            owner: The owner of the repository.
+            repo: The name of the repository.
+
+        Raises:
+            RuntimeError: If the repository is not public.
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}"
+        response = requests.get(url)
+
+        try:
+            if response.status_code == 200:
+                response.json()
+                logger.info(f"Repository {repo} is public")
+            else:
+                raise RuntimeError(f"Repository {repo} is not public")
```

Comment:
Maybe we can make the error message a little more clear by stating that a public repository is necessary when not providing a token?